tools/zetasql-helper/bigquery-antipattern-recognition/src/main/java/com/google/zetasql/toolkit/antipattern/parser/visitors/IdentifyCTEsEvalMultipleTimesVisitor.java

```diff
@@ -0,0 +1,75 @@
+package com.google.zetasql.toolkit.antipattern.parser.visitors;
+
+import com.google.zetasql.parser.ASTNodes.ASTSelect;
+import com.google.zetasql.parser.ASTNodes.ASTTablePathExpression;
+import com.google.zetasql.parser.ASTNodes.ASTWithClause;
+import com.google.zetasql.parser.ParseTreeVisitor;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class IdentifyCTEsEvalMultipleTimesVisitor extends ParseTreeVisitor {
+
+  // A string template to be used for generating the suggestion message.
+  private final String MULTIPLE_CTE_SUGGESTION_MESSAGE =
+      "Your query uses CTE alias %s which is referenced %d times.";
```

Comment:
```suggestion
      "CTE with multiple references: alias %s is referenced %d times.";
```

Reply:
Could you please add the new anti pattern to the main [here](https://github.com/meghabedi/professional-services/blob/anti-patterns/tools/zetasql-helper/bigquery-antipattern-recognition/src/main/java/com/google/zetasql/toolkit/antipattern/parser/Main.java#L50).